lib/config.rb

```diff
@@ -47,6 +48,14 @@ def self.load_files(*sources)
 
     config.add_source!(Sources::EnvSource.new(ENV)) if Config.use_env
 
+    if defined?(::Rails::Railtie) && Config.use_rails_credentials
+      if Rails.application.credentials.to_h.present?
```

Comment:
I think when some of the tests run, even on Rails 7.1, `Rails.application.credentials.to_h` will be an empty `Hash`, which will cause us to go into the `else` branch, which will access `Rails.application.secrets`, which causes a deprecation warning. Maybe we can use `respond_to` or something to avoid this from happening

```suggestion
      if Rails.application.respond_to?(:credentials)
```

This _might_ or might not be related to test failures. I think we do try to raise on deprecation warnings in the test suite.

Reply:
Thanks @cjlarose,  I did update the reset method in my stash. I'll see what else I can do. My local is using Ruby 3.2.2 with Rails 7.1 and the test pass as well on my local.